src/main/webapp/searchUsersScript.js

```diff
@@ -6,31 +6,55 @@ function searchUsersOnload() {
 }
 
 /*
- * Displays the found users from the current user's search.
+ * Finds the users whose names contain the searched text.
  */
-function displaySearchedUsers() {
-  const searchedUsers = document.getElementById('searched-users');
-  searchedUsers.style = 'height: 100%';
+function searchUsers() {
   const searchText = document.getElementById('search-text').value;
-  if (searchText === '') {
-    // Don't return results for an empty search.
-    searchedUsers.innerHTML = '';
-  } else {
+  if (searchText !== '') {
+  let searchedUsers = [];
     fetch('/user').then(response => response.json()).then((users) => {
```

Comment:
I would consider fetching all the users only once if possible. Storing them in a local array and then using that to iterate over every user. It could save a lot of fetch calls.

Reply:
In large applications, fetching the entire set of users has severe downsides. Among others:

1. It costs a lot of bandwidth, which will cost your mobile users lots of money.
2. It costs a lot of memory, which will almost certainly crash the page.
3. You may unintentionally expose usernames/profiles that shouldn't be visible to the current user (i.e., security/permissions concerns).

As a result, most real-world applications have dedicated API endpoints for search and autocomplete. For this project, I think "fetch once" and "fetch on `keyup`" are both fine.

If you are worried about fast typists slowing down your app, you can also consider debouncing your `fetch()` calls.

https://lodash.com/docs/#debounce